internal/rootcoord/meta_table.go

```diff
@@ -448,6 +450,30 @@ func (mt *MetaTable) AlterCollection(ctx context.Context, oldColl *model.Collect
 	return nil
 }
 
+func (mt *MetaTable) RenameCollection(ctx context.Context, oldName string, newName string, ts Timestamp) error {
+	mt.ddLock.RLock()
+	defer mt.ddLock.RUnlock()
+	ctx = contextutil.WithTenantID(ctx, Params.CommonCfg.ClusterName.GetValue())
+
+	oldColl, err := mt.GetCollectionByName(ctx, oldName, ts, true)
```

Comment:
I don't think this behavior is reasonable. The users shouldn't see the unavailable collections.

Reply:
Also, maybe this method will cause deadlock with `GetCollectionByName`.

Reply:
By the way, we don't check if `oldName` is an alias of collection.

Reply:
It just gets a collection meta by a collection name, it should not care about whether unavailable